pypeit/core/fitting.py

```diff
@@ -0,0 +1,1109 @@
+    minimum_useful_version = '1.0.0'
+    version = '1.0.0'
+    #
+    datamodel = {
+        'xval': dict(otype=np.ndarray, atype=np.floating, desc='x inputs'),
+        'yval': dict(otype=np.ndarray, atype=np.floating, desc='y inputs'),
+        'order': dict(otype=np.ndarray, atype=np.integer, desc='The order of the polynomial to be used in the fitting. This is a 2d array for 2d fits'),
+        'x2': dict(otype=np.ndarray, atype=np.floating, desc='x2 inputs, second independent variable'),
+        'weights': dict(otype=np.ndarray, atype=np.floating, desc='Weights'),
+        'fitc': dict(otype=np.ndarray, atype=np.floating, desc='Fit coefficients'),
+        'fitcov': dict(otype=np.ndarray, atype=np.floating, desc='Covariance of the coefficients'),
+        'gpm': dict(otype=np.ndarray, atype=np.integer, desc='Mask (1=good)'),
+        'success': dict(otype=int, desc='Flag indicating whether fit was successful (success=1) or if it failed (success=0)'),
+        'func': dict(otype=str, desc='Fit function (polynomial, legendre, chebyshev, polynomial2d, legendre2d)'),
+        'minx': dict(otype=float,
+                     desc='minimum value in the array (or the left limit for a legendre / chebyshev polynomial)'),
+        'maxx': dict(otype=float,
+                     desc='maximum value in the array (or the right limit for a legendre / chebyshev polynomial)'),
+        'minx2': dict(otype=float, desc='Same as minx for the second independent variable x2'),
+        'maxx2': dict(otype=float, desc='Same as maxx for the second independent variable x2'),
+    }
+
+    # This needs to contain all datamodel items
+    def __init__(self, xval=None, yval=None, order=None, x2=None, weights=None, fitc=None,
+                 fitcov=None, func=None, minx=None, maxx=None, minx2=None,
+                 maxx2=None, gpm=None, success=None):
+        # Setup the DataContainer
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        _d = {k: values[k] for k in args[1:]}
+        # Init
+        super(PypeItFit, self).__init__(d=_d)
+
+    def _bundle(self):
+        """
+        Bundle the data in preparation for writing to a fits file.
+
+        See :func:`pypeit.datamodel.DataContainer._bundle`. Data is
+        always written to a 'PYPEITFIT' extension.
+        """
+        return super(PypeItFit, self)._bundle(ext='PYPEITFIT')
+
+    def to_hdu(self, hdr=None, add_primary=False, primary_hdr=None,
+               limit_hdus=None, force_to_bintbl=True):
+        """
+        Over-ride :func:`pypeit.datamodel.DataContainer.to_hdu` to force to
+        a BinTableHDU
+
+        See that func for Args and Returns
+        """
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        _d = dict([(k,values[k]) for k in args[1:]])
+        # Force
+        _d['force_to_bintbl'] = True
+        # Do it
+        return super(PypeItFit, self).to_hdu(**_d)
+
+    @property
+    def bool_gpm(self):
```

Comment:
Looking back after going through this PR - I like the use of this, and it should be used in a few more places throughout. I also propose making a `bool_bpm`

Reply:
This would all be solved if we could figure out how to assign booleans in data containers.